server/src/main/java/io/crate/protocols/postgres/PostgresWireProtocol.java

```diff
@@ -272,6 +286,10 @@ private Properties readStartupMessage(ByteBuf buffer) {
         return properties;
     }
 
+    private KeyData readCancelRequestBody(ByteBuf buffer) {
+        return new KeyData(buffer.readInt(), buffer.readInt());
+    }
```

Comment:
Could make this static or move the logic into `KeyData` either as constructor `new KeyData(ByteBuf)` or as static factory `KeyData.of(ByteBuf)`. I think moving it would help with discoverability and my gut feeling is that it fits better from a responsibility perspective as well.